[ozone/wayland] occasional never ending wait at start up

tonikitoo opened the pull request:
fixup! Allow ws to pass display::Display data to Aura/client

Stop early listen to events, when wl_output is installed.
This call was causing us to enter the event loop earlier than we should,
and occasional hangs at start up.

Instead, when display/screen data is queried, we "ask" the server
to close all pending requests associated with a given wl_display.
Screen dimensions then can be queried.

Issue #212

msisov commented on ui/ozone/platform/wayland/ozone_platform_wayland.cc:
```diff
@@ -75,6 +75,10 @@ class OzonePlatformWayland : public OzonePlatform {
   void QueryHostDisplaysData(QueryHostDisplaysDataCallback callback) override {
+    // Make sure display output data is available by processing all pending
+    // requests.
+    wl_display_roundtrip(connection_->display());
```
"It is not allowed to call this function while the thread is being prepared for reading events,
 and doing so will cause a dead lock.". Maybe we better remove the previous patch we made and do the following to know the dimensions in advance?:

wl_display* display = wl_display_connect(nullptr);

static const struct wl_registry_listener registry_output = {
    OzonePlatformWayland::HandleDisplayGeometry,
};

wl_registry* registry = wl_display_get_registry(display);
wl_registry_add_listener(registry, &registry_output, this);

while(bounds_.empty()) {
  if (wl_display_roundtrip(display) < 0)
    return;
} 

// close connection and destroy display.
return bounds_;

and in OzonePlatformWayland::HandleDisplayGeometry() {
  if (strcmp(interface, "wl_output") == 0) 
    // get bounds here.
}

tonikitoo replied:
done

tonikitoo committed:
fixup! Allow ws to pass display::Display data to Aura/client

Stop early listen to events, when wl_output is installed.
This call was causing us to enter the event loop earlier than we should,
and occasional hangs at start up.

Instead, when display/screen data is queried, we "ask" the server
to close all pending requests associated with a given wl_display.
Screen dimensions then can be queried.

Issue #212

msisov reviewed (approved):
lgtm, btw
